fdmt/run.py

```diff
@@ -0,0 +1,193 @@
+    )
+
+    parser.add_argument(
+        'outfile',
+        type=str,
+        help='The file to be read written out,'
+        ' with shape (maxDT, nsamples_fdmt). Its filetype is inferred from'
+        ' the file extension, except presto-style output has no'
+        ' extension.'
+    )
+
+    parser.add_argument(
+        '--band-edges', '-e',
+        type=float, 
+        nargs=2,
+        default=None,
+        help='The lower and upper edges of the band, in MHz. If not given,'
+        ' will attempt to read from the filterbank file.'
+    )
+    
+    parser.add_argument(
+        '--maxdt', '-d',
+        type=int,
+        default=2048,
+        help='The number of time samples corresponding to the maximum delay'
+        ' between the top and bottom of the band; this defines the maximum'
+        ' DM of the search'
+    )
+
+    parser.add_argument(
+        '--verbose', '-v',
+        action='store_true',
+        help='Whether to print info to stdout.'
+    )
+
+    parser.add_argument(
+        '--padding', '-p',
+        action='store_true',
+        help='Whether to return additional, incompletely integrated time'
+        ' samples at the end of the dedispersed time series.'
+    )
+
+    parser.add_argument(
+        '--front-padding', '-f',
+        action='store_true',
+        help='Whether to return additional, incompletely integrated time'
+        ' samples at the beginning of the dedispersed time series.'
+    )
+
+    args = parser.parse_args()
+
+    fil = blimpy.Waterfall(args.infile)
+    data = np.squeeze(fil.data)
+
+    nchans = fil.header.get('nchans', data.shape[1])
+
+    if args.band_edges is None:
+        fch1 = fil.header['fch1']
+        foff = fil.header['foff']
+        tsamp = fil.header['tsamp']
+
+        if foff < 0:
+            # this is the middle of the highest frequency band
+            fmax = fch1 - 0.5 * foff   # fmax is larger than fch1
+            fmin = fmax + nchans * foff   # fmin is smaller than fmax
+            data = data[:, ::-1]
+        else:
+            fmin = fch1 - 0.5 * foff   # fmin is smaller than fch1
+            fmax = fmin + nchans * foff   # fmax is greater than fmin
+    else:
+        fmin, fmax = band_edges
+        if fmin > fmax:
+            fmin, fmax = fmax, fmin
+            data = data[:, ::-1]
+
+        print('Initializing dedispersion between',
+              fmin, 'and', fmax, 'MHz')
+
+    fdmt = FDMT(fmin=fmin, fmax=fmax,
+                nchan=fil.header.get('nchans', data.shape[1]),
+                maxDT=args.maxdt
+               )
+
+    out = fdmt.fdmt(data.T, retDMT=True, verbose=args.verbose,
+                    padding=args.padding, frontpadding=args.front_padding)
+
+    outfilename = args.outfile.rsplit('.', maxsplit=1)
+    if len(outfilename) == 1:
+        # Save .dat and .inf files like presto
+        DM_CONSTANT = 1.0 / 2.41e-4    # Need to know DM for filenames
+        dms = ((np.arange(0, fdmt.maxDT) /
```

Comment:
Could I just confirm that this is correct?  

aka for FDMT output the 0th row in the output corresponds to a shift of 0 samples between fmin and fmax, and the last row corresponds to a shift of maxDT-1?  

(as opposed to a row 0 => shift of 1 and las row => shift of maxDT)

Reply:
Yes, up to rounding errors, the FDMT output's row 0 is equal to summing the (undedispersed) input array along the frequency axis.